modin/experimental/backends/omnisci/query_compiler.py

```diff
@@ -605,29 +638,22 @@ def astype(self, col_dtypes, **kwargs):
 
     def setitem(self, axis, key, value):
         if axis == 1 or not isinstance(value, type(self)):
-            return super().setitem(axis=axis, key=key, value=value)
-
-        try:
-            result = self._setitem(axis, key, value)
-        # OmniSci engine does not yet support cases when `value` is not a subframe of `self`.
-        except NotImplementedError:
-            result = super().setitem(axis=axis, key=key, value=value)
-        return result
+            raise NotImplementedError(
+                f"OmniSci' setitem does not support such set of parameters: axis={axis}, value={value}"
+            )
+        return self._setitem(axis, key, value)
 
     _setitem = PandasQueryCompiler._setitem
 
     def insert(self, loc, column, value):
         if isinstance(value, type(self)):
             value.columns = [column]
-            try:
-                result = self.insert_item(axis=1, loc=loc, value=value)
-            # OmniSci engine does not yet support cases when `value` is not a subframe of `self`.
-            except NotImplementedError:
-                result = super().insert(loc=loc, column=column, value=value)
-            return result
+            return self.insert_item(axis=1, loc=loc, value=value)
 
         if is_list_like(value):
-            return super().insert(loc=loc, column=column, value=value)
+            raise NotImplementedError(
+                "OmniSci' insert does not support list-like values"
+            )
```

Comment:
what is the point of explicit replacements? is it for the sake of clarity, or does it have some extra value?

Reply:
1. It's easier and shorter to raise `NotImplementedError` rather than calling the super method with the appropriate arguments:
```python
raise NotImplementedError("'foo' parameter is not supported.")
# versus:
super().bar(
    fiz=fiz,
    baz=baz,
    foo=revert_original_foo_value_because_we_could_already_change_it(foo),
    axis=axis,
)
```
2. This helps us to notify the user exactly why the specific function defaulting to pandas (exception argument will be printed as a warning)